src/idl/src/annotation.c

```diff
@@ -132,27 +133,51 @@ annotate_autoid(
   return IDL_RETCODE_OK;
 }
 
-static idl_retcode_t
-annotate_optional(
-  idl_pstate_t *pstate,
-  idl_annotation_appl_t *annotation_appl,
-  idl_node_t *node)
-{
-  (void)node;
-  idl_warning(pstate, idl_location(annotation_appl),
-    "@optional not supported");
-  return IDL_RETCODE_OK;
-}
-
 static idl_retcode_t
 annotate_value(
   idl_pstate_t *pstate,
   idl_annotation_appl_t *annotation_appl,
   idl_node_t *node)
 {
-  (void)node;
-  idl_warning(pstate, idl_location(annotation_appl),
-    "@value not supported");
+  idl_type_t type;
+  const idl_const_expr_t *const_expr;
+  uint32_t value;
+
+  assert(annotation_appl);
+  assert(annotation_appl->parameters);
+  const_expr = annotation_appl->parameters->const_expr;
+
+  if (!idl_is_enumerator(node)) {
+    idl_error(pstate, idl_location(annotation_appl),
+      "@value cannot be applied to '%s' elements", idl_construct(node));
+    return IDL_RETCODE_SEMANTIC_ERROR;
+  }
+
+  type = idl_type(const_expr);
+  if (type == IDL_OCTET || (type & IDL_INTEGER_TYPE)) {
+    idl_intval_t intval = idl_intval(const_expr);
+
+    if ((intval.type & IDL_UNSIGNED) && intval.value.ullng > INT32_MAX) {
```

Comment:
Line 160: Shouldn't the check be that intval.value.llng > UINT32_MAX?

Reply:
Good catch.

Reply:
It's been resolved in the latest version. Thanks @e-hndrks!